app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -1265,22 +1265,18 @@ public void setSession(@NonNull Session aSession, @OldSessionDisplayAction int a
             mViewModel.setIsDesktopMode(mSession.getUaMode() == WSessionSettings.USER_AGENT_MODE_DESKTOP);
 
             if (hidePanel) {
-                if (oldSession != null) {
-                    onCurrentSessionChange(oldSession.getWSession(), aSession.getWSession());
-                } else {
-                    onCurrentSessionChange(null, aSession.getWSession());
-                }
+                hideNewTabPanel(true);
+                hideLibraryPanel(true);
```

Comment:
Why we don´t call here to closeLibrary instead of calling to these 2 methods  independently ?

Reply:
`closeLibrary()`  will also try to guess the next content to show, which might cause race conditions because we are in the middle of changing the session.